flax/linen/module.py

```diff
@@ -790,9 +791,52 @@ def make_rng(self, name: str) -> PRNGKey:
       raise ValueError("Can't use RNGs on unbound modules")
     return self.scope.make_rng(name)
 
+  def bind(self, variables: VariableDict, *args, rngs: RNGSequences = None,
+           mutable: CollectionFilter = False):
+    """Creates an interactive Module instance by binding variables and RNGs.
+
+    bind provides an "interactive" instance of a Module directly without
+    transforming a function with ``apply``. This is particulary useful for debugging
+    and interactive use cases like notebooks where a function would limit the ability
+    split up code into different cells.
+
+    Once the variables (and optionally RNGs) are bound to a ``Module`` it becomes a
+    stateful object. Note that idiomatic JAX is functional and therefore an interactive
+    instance does not mix well well with vanilla JAX APIs. Therefore, we recommend using
+    ``apply`` when code should be reusable and compatible across the JAX software ecosystem.
```

Comment:
Not something that should block merging this PR, but it could be nice to have a short note showcasing exactly how this fails, as I think this may be a recurring question for people new to JAX who will see `bind` and want to use it more broadly. But we can also wait to see if people ask.

Reply:
This is also something that could be part of the HOWTO